allennlp/modules/transformer/__init__.py

```diff
@@ -8,147 +8,117 @@
-encoder = TransformerEncoder.from_pretrained_module(pretrained.encoder, num_hidden_layers=4)
-
-tiny = TinyTransformer(vocab, embedding_file, embedding_dim=300, encoder=encoder)
+pretrained = cached_transformers.get("bert-base-uncased", False)
+
+class SmallTransformer(TokenEmbedder):
+    def __init__(self):
+        super().__init__()
+        self.embeddings = TransformerEmbeddings.from_pretrained_module(pretrained)
+
+        self.transformer = TransformerBlock.from_pretrained_module(
+            pretrained, num_hidden_layers=4
+        )
+
+    @overrides
+    def forward(self, token_ids: torch.LongTensor):
+        x = self.embeddings(token_ids)
+        x = self.transformer(x)
+        return x
+
+small = SmallTransformer()
+assert len(small.transformer.layers) == 4
+small.forward(torch.LongTensor([[0, 1, 2]]))
 ```
 
-Alternatively, override the `from_pretrained_module` method in `TinyTransformer`.
+3. Use the first 8 layers of `bert-base-uncased` to separately encode two text inputs, combine the representations,
+and use the last 4 layers on the combined representation.
 
 ```
-    @classmethod
-    def from_pretrained_module(
-        cls,
-        pretrained_module: torch.nn.Module,
-        vocab: Vocabulary,
-        embedding_file: str,
-        embedding_dim: int,
-        source="huggingface",
-        mapping: Optional[Dict[str, str]] = None,
-        **kwargs,
+class MediumTransformer(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.embeddings = TransformerEmbeddings.from_pretrained_module("bert-base-uncased")
+        self.separate_transformer = TransformerBlock.from_pretrained_module(
+            "bert-base-uncased", num_hidden_layers=range(0, 8)
+        )
+        self.combined_transformer = TransformerBlock.from_pretrained_module(
+            "bert-base-uncased",
+            num_hidden_layers=range(8, 12),
+        )
+
+    @overrides
+    def forward(
+        self,
+        left_token_ids: torch.LongTensor,
+        right_token_ids: torch.LongTensor,
     ):
-        encoder = TransformerEncoder.from_pretrained_module(pretrained_module.encoder, source, mapping, **kwargs)
-        final_kwargs = {}
-        final_kwargs["vocab"] = vocab
-        final_kwargs["embedding_file"] = embedding_file
-        final_kwargs["embedding_dim"] = embedding_dim
-        final_kwargs["encoder"] = encoder
-        return cls(**final_kwargs)
-
-tiny = TinyTransformer.from_pretrained_module(pretrained, vocab, embedding_file, embedding_dim=300, num_hidden_layers=4)
+
+        left = self.embeddings(left_token_ids)
+        left = self.separate_transformer(left)
+
+        right = self.embeddings(right_token_ids)
+        right = self.separate_transformer(right)
+
+        # combine the sequences in some meaningful way. here, we just add them.
+        # combined = combine_masked_sequences(left, left_mask, right, right_mask)
+        combined = left + right
```

Comment:
Maybe make it clear that we keep this simpler for clarity, but in a real world scenario you'd probably want to concatenate them. This example, as it is, will only run if the two sequences are the same length, which is a little weird.